lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -411,6 +409,9 @@ def host_to_url(h)
     def exists?(path, use_get=false)
       response = use_get ? @pool.get(path) : @pool.head(path)
       response.code >= 200 && response.code <= 299
+    rescue ::LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError => e
+      return true if e.code == 404
+      raise e
```

Comment:
should be false if code is 404, right? interesting that the tests didn't catch this.

```suggestion
      use_get ? @pool.get(path) : @pool.head(path)
    rescue ::LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError => e
      return false if e.code == 404
      raise e
    else
      true
```

Reply:
I was confused on that! I had copied that from https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1203/files#diff-4de2e59688ce60550472a447d9534d75298d14b845b5c77293f5a255359efac1R413 

Previous behavior was on 404 nothing was raised and `exists?` returned false. Your suggestion preserves this behavior. I'll look in to where we have a test hole here.